src/commonMain/kotlin/exchange.dydx.abacus/protocols/ParserProtocol.kt

```diff
@@ -71,3 +76,45 @@ inline fun <reified T> ParserProtocol.asTypedList(list: Any?): List<T>? {
         }
     }
 }
+
+inline fun <reified T> ParserProtocol.asTypedObject(item: Any?): T? {
```

Comment:
Does this need to be an extension on `ParserProtocol`?

I didn't see you had this, I wrote this in my PR:
```
inline fun <reified T> safeDecode(json: String): T? {
    return try {
        Json.decodeFromString<T>(json)
    } catch (e: SerializationException) {
        Logger.e { "Failed to decode: $json" }
        null
    } catch (e: IllegalArgumentException) {
        Logger.e { "Failed to decode: $json" }
        null
    }
}
```

Reply:
Since the code uses parser for all parsing needed, I think it's better to keep it consistent.